### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by splitbrain**
This refactors the plugin to make it easier to integrate into templates. It no longer generates icons but simply uses AJAX calls to have analyzing asynchronous. Icons are SVG now and can be styled.

For the sprintdoc template we would not use the default template function but do the two AJAX calls ourselves to integrate it into the tab interface.

@micgro42 can you have a look if I've overseen something?

### CODE REVIEW

File: action/ajax.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * @param $param
+     */
+    public function ajax(Doku_Event $event, $param) {
+        if(substr($event->data, 0, 10) != 'plugin_qc_') return;
+        $event->preventDefault();
+        $event->stopPropagation();
+        global $INPUT;
+
+        $id = cleanID($INPUT->str('id'));
+        if(blank($id)) die('no id given');
```
Comment (micgro42) [inline action/ajax.php]: We may want to do some ACL-checking here. I think `AUTH_EDIT` would make sense, so only those users, that are able to do something about it, will see the information. Also we need to check for `$this->getConf('adminonly')`.

File: Output.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        } elseif($score) {
+            $rating = 'meh';
+        } else {
+            $rating = 'good';
+        }
+
+        // output icon and score
+        $html .= '<span class="qc_icon qc_' . $rating . '">';
+        $html .= inlineSVG(__DIR__ . '/svg/' . $rating . '.svg');
+        if($score) $html .= '<span>' . $score . '</span>';
```
Comment (micgro42) [inline Output.php]: I think adding a class might make styling more descriptive. Maybe something like `qc_score`?

File: helper.php [hunk 3]
```diff
@@ -4,65 +4,56 @@
...
+    function tpl() {
+        if(!$this->shouldShow()) return;
+
         echo '<div id="plugin__qc__wrapper">';
-        echo '<img src="'.DOKU_BASE.'lib/plugins/qc/icon.php?id='.$ID.'" width="600" height="25" alt="" id="plugin__qc__icon" />';
-        echo '<div id="plugin__qc__out" style="display:none"></div>';
+        echo '<div class="summary">';
+        echo $this->getLang('i_qcscore');
+        echo '</div>';
+        echo '<div class="qc-output"></div>';
```
Comment (micgro42) [inline helper.php]: Semantically it would make probably sense to designate this as an `<aside>` instead of a `<div>`.

File: style.less [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    &.qc_good {
+        fill: #393;
+        color: #393;
+    }
+}
+
+
+div#plugin__qc__wrapper {
+    background-color: @ini_background_neu;
+    color: @ini_text_neu;
```
Comment (micgro42) [inline style.less]: This makes the Plugin very prominent in the dokuwiki template. Also, the default color combination of #666666 and #dddddd fails almost every contrast test. I would suggest using black text as the old image did.

File: style.less [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    border: 1px solid @ini__border;
```
Comment (micgro42) [inline style.less]: Typo: It should be `@ini_border`

File: Output.php (hunk 2 above)
Comment (splitbrain) [inline Output.php]: Hmm it's the only span inside .qc_icon. It's adressable via the cascade already. I don't see why a class would be necessary.

File: helper.php (hunk 3 above)
Comment (splitbrain) [inline helper.php]: good idea.

File: style.less (hunk 4 above)
Comment (splitbrain) [inline style.less]: hmm. AFAIK those colors are meant to be used in combination. So if the contrast is shitty, we should probably fix it in the default template. But yeah, styles could be optimized I guess.

File: action/ajax.php (hunk 1 above)
Comment (splitbrain) [inline action/ajax.php]: true. I think it should be added to ``helper::shouldShow()`` and that method should be used here as well.

**Pull request merged**